ludwig/schema/features/bag_feature.py

```diff
@@ -1,19 +1,20 @@
 from marshmallow_dataclass import dataclass
 
 from ludwig.constants import BAG
-from ludwig.schema import utils as schema_utils
 from ludwig.schema.encoders.base import BaseEncoderConfig
 from ludwig.schema.encoders.utils import EncoderDataclassField
 from ludwig.schema.features.base import BaseInputFeatureConfig
 from ludwig.schema.features.preprocessing.base import BasePreprocessingConfig
 from ludwig.schema.features.preprocessing.utils import PreprocessingDataclassField
-from ludwig.schema.features.utils import input_config_registry
+from ludwig.schema.features.utils import input_config_registry, input_mixin_registry
+from ludwig.schema.utils import BaseMarshmallowConfig
 
 
-@input_config_registry.register(BAG)
+@input_mixin_registry.register(BAG)
```

Comment:
Another good call. I think my issue here is not realizing I should stop, make a separate PR, merge that, and then pull that into my branch. Will stick to that workflow in the future

Reply:
That works!

Another workflow I recommend that comes without developer interruption (discussed offline) is to build a prototype of the full change in all of it's size and glory (aka this PR), and then using that as a reference to generate small, focused, maximally digestible PRs for smooth easier reviewing for the team.

Development happens as usual, but then merging in "a big PR" becomes a more logistical matter of figuring out how to best present / breakdown the technical details for your team.

- This will only be in service of others being able to appreciate the technical depth of your work, and lowering the bar for other engineers to understand, accept, and help maintain(!) your fully materialized vision (implementation, implications, etc.).
- Uncontroversial changes with a tight scope can be reviewed, merged, and rebased on quickly.
- Changes can often be reviewed and merged in parallel.
- When changes are broken down and more visible, you'll discover that sometimes there are additional test cases, comments, or other context that were missed or expanded upon, and the separate PR may turn into a more enriched/mature version than your initial reference, which you can happily merge into your reference.
- You might get push back for one of these intermediate changes that might result in needing to fundamentally change your prototype, but presumably there would be a discussion and there would be good reason for it, and it's exactly the kind of review that we want.

Reply:
Per this recommended workflow, I have factored this change out into its own PR: https://github.com/ludwig-ai/ludwig/pull/2628